NLog.Targets.OpenTelemetryProtocol/OtlpTarget.cs

```diff
@@ -140,44 +140,75 @@ private ResourceBuilder CreateResourceBuilder()
 
         protected override void Write(LogEventInfo logEvent)
         {
-            var data = new LogRecordData
+            var formattedMessage = RenderLogEvent(Layout, logEvent);
+
+            var data = new LogRecordData()
             {
                 SeverityText = logEvent.Level.ToString(),
+                Severity = ResolveSeverity(logEvent.Level),
                 Timestamp = logEvent.TimeStamp,
-                Body = logEvent.FormattedMessage
+                Body = formattedMessage,
             };
 
-            var attributes = new LogRecordAttributeList { { OriginalFormatName, logEvent.Message } };
+            var attributes = new LogRecordAttributeList();
+            AppendAttributes(logEvent, ref attributes);
+            _logger.EmitLog(data, attributes);
+        }
 
-            if (logEvent.Parameters is not null && logEvent.Parameters.Length > 0)
-                AppendParameters(logEvent, ref attributes);
+        private void AppendAttributes(LogEventInfo logEvent, ref LogRecordAttributeList attributes)
+        {
+            if (logEvent.Exception != null)
+                attributes.RecordException(logEvent.Exception);
 
-            if (_attributes.Count > 0)
+            if (logEvent.HasProperties)
             {
-                foreach (var kvp in _attributes)
+                attributes.Add(OriginalFormatName, logEvent.Message);
```

Comment:
This attribute always needs to be in the log.

As I mentioned in a comment, the logging API was originally made to work with Microsoft's ILogger, so it is built around its logic. The point is that when LogRecords are transformed into the final log format by the exporter, the logic for populating body is this:
If LogRecord.FormattedMessage is not null, then the body of the final log is set to LogRecord.FormattedMessage and the {OriginalFormat} key value pair is added to attributes. If LogRecord.FormattedMessage is null, then the body of the final log is set to the value of {OriginalFormat} and {OriginalFormat} isn't added to attributes. 
This can be seen [here](https://github.com/open-telemetry/opentelemetry-dotnet/blob/52a03b11aef3b75fdeaa4dd0fe8e9b1d71ea4e0a/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs#L150).

What this means is that if logEvent had no parameters and IncludeFormattedMessage was set to false, the body of the final log would be empty and the original format wouldn't be in attributes. 

What this also means is that LogRecord.Body itself doesn't actually do anything and that's why I have to set LogRecord.FormattedMessage in the processor if I want to include the formatted message.

I hope I'm making sense.

Reply:
Now `OriginalFormatName` is always added as attribute.